### PR TITLE
fix(core):  optimize upgrade to team page

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1922 AF-1919

fix(core): add confirm modal to remove member

fix(core): add login detection to upgrade-to-team page

fix(core): add selected recurring to upgrade-to-team page

fix(core): cancel and resume team subscription

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#9099** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9099?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9099?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/core/src/desktop/pages/upgrade-to-team/index.tsx [hunk 1]
```diff
@@ -36,6 +39,18 @@ const benefitList: I18nString[] = [
 ];
 
 export const Component = () => {
+  const authService = useService(AuthService);
+  const authStatus = useLiveData(authService.session.status$);
+  const [params] = useSearchParams();
+  const recurring = params.get('recurring');
+
+  if (authStatus === 'unauthenticated') {
+    return <PageNotFound noPermission />;
```
Comment (forehalo) [inline packages/frontend/core/src/desktop/pages/upgrade-to-team/index.tsx:50]: should redirect to sign in page

**Review by forehalo — APPROVED** [review]

File: packages/frontend/core/src/modules/cloud/entities/workspace-subscription.ts [hunk 2]
```diff
@@ -40,19 +40,37 @@ export class WorkspaceSubscription extends Entity {
   server = this.workspaceServerService.server;
   store = this.workspaceServerService.server?.scope.get(SubscriptionStore);
 
-  async resumeSubscription(idempotencyKey: string, plan?: SubscriptionPlan) {
+  async resumeSubscription(
+    idempotencyKey: string,
+    plan?: SubscriptionPlan,
+    workspaceId?: string
```
Comment (EYHN) [inline packages/frontend/core/src/modules/cloud/entities/workspace-subscription.ts]: just use `this.workspaceService.workspace.id `?

File: packages/frontend/core/src/desktop/pages/upgrade-to-team/index.tsx [hunk 3]
```diff
@@ -36,6 +39,18 @@ const benefitList: I18nString[] = [
 ];
 
 export const Component = () => {
+  const authService = useService(AuthService);
+  const authStatus = useLiveData(authService.session.status$);
+  const [params] = useSearchParams();
+  const recurring = params.get('recurring');
+
+  if (authStatus === 'unauthenticated') {
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/pages/upgrade-to-team/index.tsx]:
```suggestion
  const authIsRevalidating = useLiveData(authService.session.isRevalidating$);
  if (authStatus === 'unauthenticated' && !authIsRevalidating) {
```

File: packages/frontend/core/src/desktop/pages/upgrade-to-team/index.tsx (hunk 1 above)
Comment (JimmFly) [reply packages/frontend/core/src/desktop/pages/upgrade-to-team/index.tsx:50]: Yes, there is a login page here. After logging in, you will return to this upgrade page.

**Review by EYHN — APPROVED** [review]

**Pull request merged**